frontend/src/components/QuestionWrapper.vue

```diff
@@ -0,0 +1,166 @@
+<script setup lang="ts">
+import { computed, defineAsyncComponent } from 'vue';
+import type { ComputedRef, Component } from 'vue';
+import type { Quizoot } from '@schemas/interface';
+import NavigationButton from '@/components/NavigationButton.vue';
+import { pascalToSnake } from '@/lib/string-utils';
+
+interface QuestionsFlow {
+    questionNumber: number;
+    isFirstQuestion: boolean;
+    isLastQuestion: boolean;
+}
+
+interface QuestionWrapperProps {
+    question: Quizoot.Question;
+    questionsFlow: QuestionsFlow;
+    questionsCount: number;
+    goToPreviousQuestion: () => void;
+    goToNextQuestion: () => void;
+}
+
+type Components = Promise<any>;
+type ComponentsMap = Record<Quizoot.QuestionKind, Components>;
+
+function getComponents() {
+    const files = import.meta.glob('@/components/questions/*.vue');
+    const components: Record<string, Components> = {};
+
+    for (const filepath in files) {
+        const filename = filepath.match(/.*\/(.+)\.vue$/)?.[1];
+        if (filename) {
+            const kind = pascalToSnake(filename).toUpperCase();
+            components[kind] = defineAsyncComponent(
+                () => import(`@/components/questions/${filename}.vue`)
+            );
+        }
+    }
+
+    return components as ComponentsMap;
+}
+
+const components = getComponents();
+
+const props = defineProps<QuestionWrapperProps>();
+
+const QuestionComponent: ComputedRef = computed(
```

Comment:
You changed this. Why ?

Previously, it was :
```vue
const componentId = computed(() => {
    if (props.question.kind in components) {
        return components[props.question.kind];
    }
    throw Error(`Unknown QuestionKind "${props.question.kind}"`);
});
```

## Notes
In your current implementation, you don't need to add `as Component` at the end of your function, since the `components` object is already typed. You should even avoid doing that.

Reply:
If I understand, when an element is already typed somewhere, there is no need to specify the type afterwards ?

Reply:
The components async loading was not working, and I deleted those lines while debugging. Thought I restored them, as they were not the problem.

Reply:
Got it now, thanks! 😊
And you're right, typescript complained when I changed the JSON data a bit.